File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    n1:=len(haystack)
    n2:=len(needle)
    if n1<n2{
        return -1
    }
    // for i=0;i<n1-n2;i++{
    //     s:=
    // }
    return strings.Index(haystack, needle)
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    var strr;
    for(let i=0; i < haystack.length;i++){
        strr = haystack.substring(i, needle.length+i);
        if(strr===needle)return i;
    }
    return -1;
};